### PR TITLE
Mark scale as const and remove --fp8 flag usage

### PR DESCRIPTION
**Pull request by Yantom1**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: examples/text-generation/utils.py [hunk 1]
```diff
@@ -367,7 +367,7 @@ def initialize_model(args, logger):
     set_seed(args.seed)
     get_repo_root(args.model_name_or_path, local_rank=args.local_rank, token=args.token)
     use_deepspeed = args.world_size > 0
-    if use_deepspeed or args.bf16 or args.fp8:
```
Comment (HolyFalafel) [inline examples/text-generation/utils.py:370]: This changes behavior than what we had before, is it tested on 7b? (seems like the only way to be false)

File: examples/text-generation/utils.py [hunk 2]
```diff
@@ -138,7 +138,7 @@ def setup_device(args):
     if args.device == "hpu":
         import habana_frameworks.torch.core as htcore
 
-        if args.fp8:
+        if args.quant_config:
```
Comment (HolyFalafel) [inline examples/text-generation/utils.py:141]: This is true also when performing a measurement, was it tested?

File: examples/text-generation/utils.py [hunk 3]
```diff
@@ -391,7 +391,7 @@ def initialize_model(args, logger):
 
     if args.const_serialization_path:
         setup_const_serialization(args.const_serialization_path)
-    if args.fp8:
+    if args.quant_config:
```
Comment (HolyFalafel) [inline examples/text-generation/utils.py:394]: Also in measurement

**Review by HolyFalafel — CHANGES_REQUESTED** [review]

File: examples/text-generation/utils.py (hunk 3 above)
Comment (Yantom1) [reply examples/text-generation/utils.py:394]: yes

File: examples/text-generation/utils.py (hunk 2 above)
Comment (Yantom1) [reply examples/text-generation/utils.py:141]: yes

File: examples/text-generation/utils.py [hunk 4]
```diff
@@ -102,7 +102,7 @@ def setup_inference(args, model):
     print("Initializing inference mode")
     const_marking = os.getenv("ENABLE_CONST_MARKING", "True")
```
Comment (MrGeva) [inline examples/text-generation/utils.py:103]: @Yantom1 you should remove it. we should not use this variable anymore. we should always call hpu_initialize in this function. as QA set this flag to false, did you test it correctly?

**Comment by astachowiczhabana** [comment]
https://github.com/huggingface/optimum-habana/pull/962

**Pull request merged**